src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,14 +1,33 @@
 package core.basesyntax.service;
 
+import core.basesyntax.Exception.NoValidDataException;
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        char []loginLength = user.getLogin().toCharArray();
+        char []passwordLength = user.getPassword().toCharArray();
```

Comment:
It's unnecessary to convert login and password to char arrays just to check their lengths. You can directly use `user.getLogin().length()` and `user.getPassword().length()`.

Reply:
It's unnecessary to convert login and password to char arrays just to check their lengths. You can directly use user.getLogin().length() and user.getPassword().length().